### ISSUE TITLE
Can Opensuse Leap be added to the supported distributions

### ISSUE THREAD

**Issue by MortalCatalyst** [nodesource/distributions#199]
Hi 

Just wondering if opensuse Leap can we added to the list of supported rpm distributions?

If it is and i am using the wrong installed forgive me.

```bash
sayth@linux-l8lq:~> curl -sL https://rpm.nodesource.com/setup_5.x | bash -

## Installing the NodeSource Node.js 5.x repo...


## Inspecting system...

+ rpm -q --whatprovides redhat-release || rpm -q --whatprovides centos-release || rpm -q --whatprovides cloudlinux-release || rpm -q --whatprovides sl-release
+ uname -m

## You don't appear to be running a supported version of Enterprise Linux. Please contact NodeSource at https://github.com/nodesource/distributions/issues if you think this is incorrect or would like your architecture to be considered for support. Include your 'distribution package' name: no package provides redhat-release
no package provides centos-release
no package provides cloudlinux-release
no package provides sl-release. 

sayth@linux-l8lq:~>

**Comment by rvagg** [comment]
I had a bit of a play with adding opensuse support a few months back but it turned out to be more difficult than just adding aliases for our existing rpms, the ecosystem is different enough that we may have to add separate a opensuse repository to make it work. I may be wrong, but it's certainly not as easy as the all EL/CentOS variants which really just build on EL.

**Comment by MortalCatalyst** [comment]
Morning

Thanks for having a look. Might see if i can use the current build in OBS
for clues on how can make it work.

Guess the good is that opensuse leap is long term supported not a target
moving all the time.

Thanks

Sayth

**Comment by raulgd** [comment]
Hi guys,

I'm also interested in this, OpenSUSE Leap is having better support running as a Docker image than CentOS (7 is having a few issues running dockerized).

Will you guys continue on this, or is there a way we could somehow help to get this moving?

Sayth, did you find something by checking OBS? the latest rpm there is node 5.9.1

http://download.opensuse.org/repositories/devel:/languages:/nodejs/openSUSE_Leap_42.1/

**Comment by rvagg** [comment]
Not something that is on our radar, as per my last comment, I has a quick attempt at doing this but soon discovered how different openSUSE's packaging environment is. Also, I don't know anything about Leap, so I'm not sure if there is anything special there that changes anything for us.

To move ahead with support, what we would probably need is a bit of handholding by someone with more in-depth knowledge about how it all works and how we might be able to adapt our packaging scripts to work with openSUSE. If you have a look in https://github.com/nodesource/distributions/tree/master/rpm you'll see how we do it for EL variants (dev work is done in src and build.sh makes all of the scripts for the different Node versions). If someone could take that and show what needs to be modified to make openSUSE we might be able to find a way forward. For other RPM platforms we've just been able to map their versions to EL versions and the RPMs we have work fine. If there's a simple mapping that we can implement for openSUSE then that's great, if there's a bit more scripting work around installing an RPM then that's doable too. If it involves changing the RPMs themselves then it'll probably be a bit trickier.

Basically, we don't have a good understanding of the openSUSE packaging system or what the delta is between it and what we are already doing. If somebody, or bodies, could help us understand then we might be able to move forward. Note though, that we're unlikely to actually test any resultant scripts ourselves on openSUSE machines so we'd need help verifying and reporting problems back to us if they emerge.

**Comment by raulgd** [comment]
Ok, I'll contact someone I know on the OpenSUSE community to help out, also Leap is actually easier to get it working than the 13.x series, because it's now based on SUSE Linux Enterprise (think of Leap as the CentOS for the RHEL) so it uses RPMs as well, the difference is just some paths differ from RHEL, so it should be to just adapt to that.

And don't worry about zypper/yum differences, in the end they are just managers for RPM underneath.

**Comment by retrohacker** [comment]
Hey @raulgd and @MortalCatalyst!

I'm moving this to our README under `Request Distributions`. I'm closing this issue to keep our issues page tidy, but I'll be linking to here from the README. Feel free to keep us posted through this issue :smile: it will land in #308 

Thank you again for championing OpenSUSE in these repos!

**Comment by DaAwesomeP** [comment]
There is an official NodeJS repository for the current versions: https://nodejs.org/en/download/package-manager/#opensuse-and-sle. The latest LTS version is listed in the OpenSUSE main OSS repository. I'm not sure that another repository is necessary.
